routes/campgrounds.js

```diff
@@ -0,0 +1,109 @@
+const express = require("express");
+const router = express.Router();
+const Campground = require("../models/campground");
+const Comment = require("../models/comment");
+const middleware = require("../middleware");
+
+router.get("/campgrounds", function (req, res) {
+  //get all campground from
+  Campground.find({}, function (err, allCampgrounds) {
+    if (err) {
+      console.log(err);
+    } else {
+      res.render("campgrounds/index", { campgrounds: allCampgrounds });
+    }
+  });
+});
+
+// Create
+router.post("/campgrounds", middleware.isLoggedIn, function (req, res) {
+  // get data from form
+  var name = req.body.name;
+  var price = req.body.price;
+  var image = req.body.image;
+  var desc = req.body.description;
+  var author = {
+    id: req.user._id,
+    username: req.user.username,
+  };
+  var newCampground = {
+    name: name,
+    price: price,
+    image: image,
+    description: desc,
+    author: author,
+  };
+
+  Campground.create(newCampground, function (err, newlyCreated) {
+    if (err) {
+      console.log(err);
+    } else {
+      res.redirect("/campgrounds");
+    }
+  });
+});
+
+// New
+router.get("/campgrounds/new", middleware.isLoggedIn, function (req, res) {
+  res.render("campgrounds/new");
+});
+
+//SHOW
+router.get("/campgrounds/:id", function (req, res) {
+  Campground.findById(req.params.id)
+    .populate("comments")
+    .exec(function (err, foundCampground) {
+      if (err) {
+        console.log(err);
+      } else {
+        console.log(foundCampground);
+        res.render("campgrounds/show", { campground: foundCampground });
+      }
+    });
+});
+
+//Edit Route
+router.get(
+  "/campgrounds/:id/edit",
```

Comment:
Instead of ‘edit’ segment from the route, maximise http methods. Example, for editing/updating data, use PUT method.